### PR TITLE
added publisher and logger package

### PR DESCRIPTION
**Pull request by Nikhil-dot-Kumar**
Checklist:

- [x] I have read [contribution guidelines](https://github.com/maplelabs/github-audit/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] I have tested these changes locally on my machine.

##### SUMMARY

1. Added logger package which provides logging functionality.
2. Added publisher package which sends data to targets.

##### COMPONENT NAME

[Publisher](https://github.com/maplelabs/github-audit/blob/master/docs/design/software_design.md#publisher)

##### ADDITIONAL INFORMATION

Current implementation is not based on original design as Tasks are already running in goroutines so publisher instances need 
not run in multiple goroutines.

### CODE REVIEW

File: logger/logger.go [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+)
+
+// logger instance
+const (
+	InstanceZapLogger int = iota
+)
+
+// default config log timestamp
+const (
+	TimestampFormat    = "15:04:05.999 02/01/2006 (MST)"
```
Comment (upendrasahu) [inline logger/logger.go:39]: why MST timezone?

File: publisher/elasticsearch.go [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+	Index string `yaml:"index" json:"index"`
+
+	// Username if basic auth is used
+	Username string `yaml:"user,omitempty" json:"user,omitempty"`
+
+	// Password if basic auth is used
+	Password string `yaml:"password,omitempty" json:"password,omitempty"`
+
+	// ES_7x to know if target is elastic version 7.x
+	ES_7x bool `yaml:"es_7x" json:"es_7x"`
```
Comment (upendrasahu) [inline publisher/elasticsearch.go:38]: Do you need this as input? check if you can query to ES to get version and set this flag. sf-trace-server is likely getting the version on the fly.

File: logger/logger.go (hunk 1 above)
Comment (Nikhil-dot-Kumar) [reply logger/logger.go:39]: changing it to UTC

File: publisher/elasticsearch.go (hunk 2 above)
Comment (Nikhil-dot-Kumar) [reply publisher/elasticsearch.go:38]: will add logic to get version in code itself

File: publisher/elasticsearch.go [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+}
+
+// BulkResponse ...
```
Comment (upendrasahu) [inline publisher/elasticsearch.go:42]: add proper comments in all structs

File: publisher/elasticsearch.go [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+	err = json.Unmarshal(respBody, &esError)
+	if err != nil {
+		log.Errorf("error[%v] failed to Unmarshal response body, %s", err)
+		return errors.New(response.Status)
+	}
+	log.Errorf("error[%v]", esError)
+	return fmt.Errorf("status code: %d, error: %s", esError.Status, esError.Error.Reason)
+}
+
+// getURL returns url to elasticendpoint for writing data
```
Comment (upendrasahu) [inline publisher/elasticsearch.go:206]: Check the elastic beats code
https://github.com/elastic/beats/tree/main/libbeat/outputs

There may be some cases like index does not exist or generating index name based on some fields on doc.
Same for Kafkarest

Comment (upendrasahu) [reply publisher/elasticsearch.go:206]: https://github.com/snappyflow/beats/tree/master/libbeat/outputs

**Review by upendrasahu — APPROVED** [review]

**Pull request merged**